sirfilesystem.c

```diff
@@ -157,6 +157,75 @@ char* _sir_getcwd(void) {
 #endif
 }
 
+#if defined(__OpenBSD__)
+static int
+_sir_openbsdself(char* out, int capacity, int* dirname_length)
+{
+  char   buffer1[4096];
+  char   buffer2[PATH_MAX];
+  char   buffer3[PATH_MAX];
+  char** argv     = (char**)buffer1;
+  char*  resolved = NULL;
+  int    length   = -1;
+
+  for (;;) {
+    int mib[4] = { CTL_KERN, KERN_PROC_ARGS, getpid(), KERN_PROC_ARGV };
+    size_t size;
+
+    if (sysctl(mib, 4, NULL, &size, NULL, 0) != 0) break;
+    if (size > sizeof(buffer1)) {
+      argv = (char**)malloc(size);
+      if (!argv) break;
+    }
+
+    if (sysctl(mib, 4, argv, &size, NULL, 0) != 0) break;
+    if (strchr(argv[0], '/')) {
+      resolved = realpath(argv[0], buffer2);
+      if (!resolved) break;
+    } else {
+      const char* PATH = getenv("PATH");
+      if (!PATH) break;
+      size_t argv0_length = strlen(argv[0]);
+      const char* begin = PATH;
+      while (1) {
```

Comment:
I would suggest either using `for (;;)` or `while (1)` in both spots to be consistent

Reply:
Easy enough.